src/test/test_cfg_builder_passes.cpp

```
@@ -207,6 +208,31 @@ TEST_CASE("pass_insert_termination_counters is off by default", "[passes]") {
+
+    SECTION("counts nested local calls") {
+        InstructionSeq seq;
+        // entry -> call subprogram 1 -> call subprogram 2 -> exit.
+        seq.push_back(at(0, CallLocal{.target = Label{2}}));
+        seq.push_back(at(1, Exit{}));
+        seq.push_back(at(2, CallLocal{.target = Label{4}}));
+        seq.push_back(at(3, Exit{}));
+        seq.push_back(at(4, Exit{}));
+
+        const Program prog = Program::from_sequence(seq, info, {});
+        REQUIRE(analyze(prog, {}).max_call_depth == 2);
```

Cppcheck (2.21.0)

[performance] Function parameter 'premise_set' should be passed by const reference.

(passedByValue)